GameProject/Assets/TurnBasedAssets/Scripts/CubeGrower.cs

```diff
@@ -0,0 +1,19 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using TurnBasedAssets.Scripts;
+using UnityEngine;
+
+public class CubeGrower : MonoBehaviour, ISelection
+{
+    [SerializeField] private Transform cubeSize;
+    [SerializeField] private const int size = 3;
```

Comment:
You cannot serialize a `const`. If you want it configurable you need to drop the `const` keyword.